### PR TITLE
Adds a simple sample app written in go

### LINKED ISSUE #18: Create sample apps to be used
For the different workflows, at least on app to be pushed is required. Setup some easy to install sample apps in languages that tend to install quickly, like Go or PHP.

Think about shipping options for those apps. Check whether there is a way if we can ship example apps inside the main program binary (like a self-extracting file carries additional payload). This would simplify the installation requirements.

### PR DESCRIPTION
**Pull request by LynxPlay101**
This PR adds a really simple testing app written in go, that will log the systems Unix timestamp to the console as well as return a JSON sample on calls against '/'. 

To prepare for future assets, this PR will also introduce the mappings.properties file which can be used to register new mappings without touching the build script every time.

Fixes #18

### CODE REVIEW

File: cf-apps/go/main.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
+// Copyright © 2019 The Homeport Team
```
Comment (HeavyWombat) [inline cf-apps/go/main.go]: Like discussed, move this file into a new directory in the root of the project called `assets`.

**Comment by HeavyWombat** [comment]
@LynxPlay101 I think the proposed setup here does not check whether a `pina-golada` binary is installed on the local machine. For that matter, it would fail to build the binaries in Travis, too. This did not surface in the tests, because we only run `make test` to do unit tests. If we were to add `build` as well, we _should_  see Travis fail.

Two things:
- Verify that we can have a breaking Travis build when adding `build` to the `make analysis test` line in the Travis configuration.
- Add a check for the required tool in `build.sh`. At the beginning, there is already boiler plate code to check for required tools.

File: internal/dom/assets/assets.go [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package assets
+
+import "github.com/homeport/pina-golada/pkg/files"
+
+//Provider is the provider instance to access pina-goladas asset framework
```
Comment (HeavyWombat) [inline internal/dom/assets/assets.go]: I do love that space after the double slash :)

File: internal/dom/assets/assets.go [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+var Provider ProviderInterface
+
+/*
+ProviderInterface is the interface that provides the asset file
+@pgl(package=assets&injector=Provider)
+ */
+type ProviderInterface interface {
+	/*
+	GetGoSampleApp returns the directory containing the go sample app
+	@pgl(asset=/assets/go-cf-sample/&compressor=tar)
```
Comment (HeavyWombat) [inline internal/dom/assets/assets.go]: Is `compressor=tar` just the tarring or does it _compress_ the content with GZip or similar as well?

File: scripts/build.sh [hunk 4]
```diff
@@ -42,6 +42,9 @@ go mod verify
 # GOOS options: darwin dragonfly freebsd linux nacl netbsd openbsd plan9 solaris windows
 # GOARCH options: 386 amd64 amd64p32 arm arm64 ppc64 ppc64le mips mipsle mips64 mips64le s390x
 #
+echo -e "\\n\\033[1mCompiling assets:\\033[0m"
+pina-golada generate
```
Comment (HeavyWombat) [inline scripts/build.sh]: The `pina-golada` is a required binary and should be checked upfront.

File: internal/dom/assets/assets.go (hunk 3 above)
Comment (LynxPlay101) [reply internal/dom/assets/assets.go]: the tar implementation does use gzip to compress it yes

**Comment by HeavyWombat** [comment]
@LynxPlay101 Could you please have a look at a rebase to deal with the merge conflicts? After that, this can be merged. I had a look at resolving the conflicts myself, but it will require a bit more insight into the names and changes. Thanks in advance.

**Pull request merged**